### PR TITLE
[ptq] Introduce fairseq encoder layer wrapper

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit introduces a wrapper for fairseq encoder layer.

TICO-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

File: test/quantization/ptq/wrappers/fairseq/test_quant_encoder_layer.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+    # 2) Mask handling: boolean causal attn_mask and boolean key_padding_mask
+    def test_mask_handling(self):
+        layer, _ = self._build_layer(True, False, QuantConfig())
+        x = self._make_inputs()
+        # Upper-triangular causal mask [T,S] (True = masked)
+        attn_mask_bool = torch.triu(
+            torch.ones(self.T, self.T, dtype=torch.bool), diagonal=1
+        )
+        # Key padding mask [B,S] (True = pad)
```
Comment (dayo09) [inline test/quantization/ptq/wrappers/fairseq/test_quant_encoder_layer.py:141]: Q. About abbreviation `T` and `S`, it seems both refers to sequence length but what's difference?

**Review by jinevening — APPROVED** [review]
LGTM

BTW, have you tested whether the exported circle is compiled by npu compiler?

File: test/quantization/ptq/wrappers/fairseq/test_quant_encoder_layer.py (hunk 1 above)
Comment (mhs4670go) [reply test/quantization/ptq/wrappers/fairseq/test_quant_encoder_layer.py:141]: `T` refers to target sequence length and `S` refers to source sequence length. 

When it comes to the comments in the codes, `attn_mask` has shape (T, S), where T = query length and S = key length. And, `key_padding_mask` is (B, S), since it marks padding positions in the key sequence for each batch.

**Comment by mhs4670go** [comment]
> BTW, have you tested whether the exported circle is compiled by npu compiler?

All I pushed have tested with npu compiler. You can refer to my personal repo's issue number 18.

**Review by dayo09 — APPROVED** [review]
LGTM, thanks for the explanation!
